### ISSUE TITLE
how to distill from CLIP to get a tiny model?

### ISSUE THREAD

**Issue by dragen1860** [openai/CLIP#72]
Dear all:
  I want to retain a smaller model from CLIP network. However, when i try to run knowledge distillation from Vit32model, I found `model.encode_image(image)` get a vector of lenght 512, while `model(image_feature, text_feature)` network need the text input. I donot know how to get the soft label from CLIP. Anyone get some idea about how to distill knowledge from CLIP? thanks .

**Comment by jongwook** [comment]
I guess there are different strategies depending on your goals. If you want to make a smaller vision model that encodes image into the feature space similar to CLIP's, so that you can use CLIP's robust representation with the faster encoder, I'd train the student encoder to minimize L2 or cosine distances to CLIP's features.

If your goal is to make a specific zero-shot image classifier faster with a predetermined set of labels, you can train your image encoder to minimize the KL divergence from the probability distribution resulting from the logits returned by the CLIP model.

**Comment by dragen1860** [comment]
@jongwook Thank you for the detailed tips. I finally get what you mean. Have you test training a mobile backbone using CLIP setting. Is it possible to get a reasonble zero-shot performance?

**Comment by jongwook** [comment]
We don't have results on smaller model sizes than the released ones, and we haven't experimented CLIP on mobile devices.  ResNet50 or ViT-B/32 model requires only a few gigaflops for the forward pass, and you might not need distillation at all if it's fine for your use case.

**Comment by dimitry12** [comment]
@dragen1860 have you tried distilling CLIP's image-encoder? If yes, which one (ViT or RN based)?

**Comment by ZJJHYM** [comment]
Hello, Can you share me the weight after distill or the code about distill the clip model?  My email is 13816267895@163.com
Thank you very much!

**Comment by justlike-prog** [comment]
@dragen1860 did you manage to make a succesful distillation?
